### ISSUE TITLE
vm_merge_need_free_disk_space despite free disk space more than 20%

### ISSUE THREAD

**Issue by tomglaza** [VictoriaMetrics/VictoriaMetrics#1682]
**Describe the bug**
I have two nodes with independent installation of Victoriametrics (first one version 1.66.2, and second 1.63.0). Both have the same configuration, and are installed on Centos7. Victoriametrics storage dir is located on raid10. In both cases used space is similar and looks like this:
```bash
/dev/sdb        893G  660G  234G  74% /var/lib/victoria-metrics-data
```

So there is more than 20% of free disk space. Unfortunately graph of vm_merge_need_free_disk_space looks like

For version 1.63.0
![image](https://user-images.githubusercontent.com/91344754/136369717-cdaf216b-8c0d-459e-9d6f-6ed8169d5311.png)
```bash
# victoria-metrics-prod -version
victoria-metrics-20210715-110943-tags-v1.63.0-0-g61cc13c16
# df -h | grep victoria
/dev/sdb        894G  684G  211G  77% /var/lib/victoria-metrics-data
# du -khs /var/lib/victoria-metrics-data/data/small/*
1.4G	/var/lib/victoria-metrics-data/data/small/2021_08
5.1G	/var/lib/victoria-metrics-data/data/small/2021_09
15G	/var/lib/victoria-metrics-data/data/small/2021_10
0	/var/lib/victoria-metrics-data/data/small/snapshots
# du -khs /var/lib/victoria-metrics-data/data/big/*
205G	/var/lib/victoria-metrics-data/data/big/2021_08
195G	/var/lib/victoria-metrics-data/data/big/2021_09
53G	/var/lib/victoria-metrics-data/data/big/2021_10
0	/var/lib/victoria-metrics-data/data/big/snapshots
```

For version 1.66.2
![image](https://user-images.githubusercontent.com/91344754/136369792-8f54482f-8a15-4492-948b-275a6edb9789.png)
```bash
# victoria-metrics-prod -version
victoria-metrics-20210923-195814-tags-v1.66.2-0-g9f1964967
# df -h | grep victoria
/dev/sdb        893G  660G  233G  74% /var/lib/victoria-metrics-data
# du -khs /var/lib/victoria-metrics-data/data/small/*
1.7G	/var/lib/victoria-metrics-data/data/small/2021_08
0	/var/lib/victoria-metrics-data/data/small/2021_09
11G	/var/lib/victoria-metrics-data/data/small/2021_10
0	/var/lib/victoria-metrics-data/data/small/snapshots
# du -khs /var/lib/victoria-metrics-data/data/big/*
172G	/var/lib/victoria-metrics-data/data/big/2021_08
213G	/var/lib/victoria-metrics-data/data/big/2021_09
55G	/var/lib/victoria-metrics-data/data/big/2021_10
0	/var/lib/victoria-metrics-data/data/big/snapshots
```

**Expected behavior**
vm_merge_need_free_disk_space at value 0 for type="storage/big" if more than 20% of disk space is available?

**Screenshots**
![image](https://user-images.githubusercontent.com/91344754/136374930-70826a90-4496-4290-9e74-9cb370a8dd46.png)

**Used command-line flags**
```
/usr/bin/victoria-metrics-prod -storageDataPath=/var/lib/victoria-metrics-data -retentionPeriod=2
```

**Comment by valyala** [comment]
The `vm_merge_need_free_disk_space` metric becomes non-zero if there are data parts, which cannot be merged because the summary size of these parts exceeds the remaining free disk space. For example, if there are two data parts, which occupy 26% of disk space individually, then these parts would occupy 26*2=52% of disk space in summary. Unfortunately these parts cannot be merged into bigger part, since VictoriaMetrics needs additional 52% of disk space for temporary files during merge process. That's why the `vm_merge_need_free_disk_space` metric will be set to non-zero value. If disk space is increased by more than 4%, then this will unblock the merge process for 26% parts.

**Comment by valyala** [comment]
The `2021_08` month contains `24+25+25+25+25+7.5+19+24=174.5GB` of data. Indeed, these parts can be merged, since there is 219GB of free space on disk with VictoriaMetrics data according to `df -h` output. But VictoriaMetrics can run multiple big merges in parallel, so it divides the free disk space by the maximum number of parallel big merges when calculating the maximum disk space a single big merge can occupy. The max number of parallel big merges is determined automatically as `CPUs_count / 2` - see [this line in the source code](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/3f49bdaeffa00888e3233453d4bc00630aa5910d/lib/storage/partition.go#L851). This number can be adjusted via `-bigMergeConcurrency` command-line flag.

**Comment by hagen1778** [comment]
Hey @valyala! There was another case in the VM's slack related to this issue. The metric for deferred merges was constantly up and merges started only after changing `-bigMergeConcurrency` to `1`. The amount of free disk space was about 4.2TB, while the `data/big` folder was only 1.7TB. Right after `-bigMergeConcurrency` was changed - big parts merges started and the metric went down.

**Comment by ns-jtenberge** [comment]
We also see this issue in VM 1.81.2

**Comment by valyala** [comment]
FYI, we decided dropping the `vm_merge_need_free_disk_space` metric, since it confuses users and has no any useful application. See https://github.com/VictoriaMetrics/VictoriaMetrics/issues/686#issuecomment-1733844128 for details.

**Comment by valyala** [comment]
FYI, VictoriaMetrics stops exposing `vm_merge_need_free_disk_space` metric starting from [v1.94.0](https://docs.victoriametrics.com/CHANGELOG.html).
